src/v2/Utils/Hooks/useScrollTo.ts

```diff
@@ -47,5 +47,5 @@ export const useScrollTo = ({
     window.scrollTo({ top: position, behavior })
   }
 
-  return { scrollTo }
+  return { scrollTo, isMatchMediaParsed: isMobile !== null }
```

Comment:
What's this mean?

Reply:
/ why would we include it in this hook? Doesn't seem relevant to scrolling.

Reply:
If we try to use `scrollTo` in `useEffect` hooks, we get that `isMobile` will be `null` and for this reason there will be an [incorrect offset](https://github.com/artsy/force/blob/dfc405fdc09032d01f881e1b21565451cb5cdb3d/src/v2/Utils/Hooks/useScrollTo.ts#L45) for mWeb

Reply:
Could you add a comment explaining that? Is there a way to rename this variable such that it's not concerned with the internal implementation details?

Reply:
> Is there a way to rename this variable such that it's not concerned with the internal implementation details

will the name `isReady` be suitable?

Reply:
Why don't we just set the default value in useMatchMedia to `false` rather than having what is essentially a 3-state boolean? It's not immediately clear to me why this is happening? The matchMedia event runs after you want it to?

Reply:
I think it is possible to refactor `__internal__use of matchMedia` hook so that the `matches` value is specified immediately and [not in `useEffect`](https://github.com/artsy/force/blob/a1d30f7aa4430b09496844f8e07183dd013f2b84/src/v2/Utils/Hooks/useMatchMedia.ts#L36).

Reply:
If you see an angle on that, by all means fix the root cause. Though it does need to be in useEffect presumably because this can only run on client-side.

Reply:
> this can only run on client-side

Good point. I didn't think about it

Reply:
> Why don't we just set the default value in useMatchMedia to false

In this case we will get the wrong behavior. For example, a user will open Force on a mobile device -> we will pass `themeProps.mediaQueries.xs` in `__internal__useMatchMedia`. In this case, we will immediately get `isMobile = false`, which is _not entirely true_, because the user uses a mobile device

Reply:
In the current implementation:
- If `__internal__useMatchMedia` returned `null`, then the `mediaQueryString` check has not yet been performed
- If `__internal__useMatchMedia` returned `true`, then the device falls under the conditions of `mediaQueryString`
- If `__internal__useMatchMedia` returned `false`, then the device is **not** under the conditions of `mediaQueryString`